editoast/src/schema/timetable.rs

```diff
@@ -0,0 +1,62 @@
+use super::{TrainSchedule, TrainScheduleDetails};
+use crate::error::{InternalError, Result};
+use crate::tables::osrd_infra_timetable;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use diesel::result::Error as DieselError;
+use editoast_derive::EditoastError;
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Debug, PartialEq, Queryable, Identifiable, Serialize, Selectable)]
+#[diesel(table_name = osrd_infra_timetable)]
+pub struct Timetable {
+    pub id: i64,
+    pub name: String,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "timetable")]
+pub enum TimetableError {
+    #[error("Timetable '{timetable_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { timetable_id: i64 },
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+pub struct TimetableWithSchedules {
+    pub id: i64,
+    pub name: String,
+    pub train_schedules: Vec<TrainScheduleDetails>,
+}
```

Comment:
This should be moved to the view.

```suggestion
#[derive(Debug, PartialEq, Serialize)]
pub struct TimetableWithSchedules {
    #[serde(flatten)]
    pub timetable: Timetable,
    pub train_schedules: Vec<TrainScheduleDetails>,
}
```